### ISSUE TITLE
Struct field lifetime error and misleading error message.

### ISSUE THREAD

**Issue by nulldatamap** [rust-lang/rust#20491]
The following code results in a misleading error message and is not intended behaviour.
```rust
pub struct Wowser<'a> {
  wow : &'a mut str,
}

impl<'a> Wowser<'a> {
  pub fn new( wowling : &'a mut str ) -> Wowser<'a> {
    Wowser{ wow: wowling }
  }
}
```
Error:
```
error: mismatched types: expected `&mut str`, found `&'a mut str` (values differ in mutability)
     Wowser{ wow: wowling }
                  ^~~~~~~
```
Passing a new lifetime to the function will result in the same error:
```rust
pub fn new<'b>( wowling : &'b mut str ) -> Wowser<'b> {
```

**Comment by steveklabnik** [comment]
This example now compiles.
